### PR TITLE
Add error validation for main forms

### PR DESCRIPTION
**Pull request by guiman**
Adding Reform as a validation mechanism for al objects

### CODE REVIEW

File: pivotal/lib/sprint_review.rb [hunk 1]
```diff
@@ -18,11 +20,19 @@ def initialize(remaining_weeks: raise, team_members: raise, iteration: raise, ch
...
       @pivotal_operations = pivotal_operations
-      @iterations = @pivotal_operations.iteration(@iteration_number)
       @change_from_planning = change_from_planning
       @retro_actions = retro_actions
       @worked_hours = worked_hours
       @planning_actions = planning_actions
+      @validator = SprintReviewContract.new(self)
+
+      if valid?
+        @iterations = @pivotal_operations.iteration(@iteration_number)
```
Comment (szalansky) [inline pivotal/lib/sprint_review.rb]: `@pivotal_operations.iteration` returns a single object and you assign it to a collection `@iterations`? or it's just a typo somewhere?

File: pivotal/lib/sprint_review.rb [hunk 2]
```diff
@@ -18,11 +20,19 @@ def initialize(remaining_weeks: raise, team_members: raise, iteration: raise, ch
       @team_members = team_members
       @iteration_number = iteration
       @pivotal_operations = pivotal_operations
-      @iterations = @pivotal_operations.iteration(@iteration_number)
       @change_from_planning = change_from_planning
       @retro_actions = retro_actions
       @worked_hours = worked_hours
       @planning_actions = planning_actions
+      @validator = SprintReviewContract.new(self)
```
Comment (szalansky) [inline pivotal/lib/sprint_review.rb]: wouldn't it be nice to inject validator as a dependency? just like other objects

Comment (szalansky) [inline pivotal/lib/sprint_review.rb]: I didn't realise the relationship between object and its validator.

File: pivotal/lib/sprint_review.rb (hunk 1 above)
Comment (guiman) [inline pivotal/lib/sprint_review.rb]: actually, I think I need to change the name of the method to reflect the result better

**Comment by guiman** [comment]
Update:

* Update test for Configuration to start passing
* Rename Operations.interation into iterations

**Pull request merged**